### PR TITLE
Add > Steampipe Cloud > Integrations Doc > Apache Superset

### PR DESCRIPTION
**Pull request by RahulSrivastav14**

### CODE REVIEW

File: docs/cloud/integrations/apachesuperset.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  Port:               9193
+  Database:           steampipe
+  User:               steampipe
+  Password:           99**_****_**8c
+  Connection string:  postgres://steampipe:99**_****_**8c@localhost:9193/steampipe
+  ```
+
+## Getting started
+
+[Apache Superset](https://superset.apache.org/docs/installation/) is a free to use open-source application that runs in a container, or on the desktop. Here we run it locally through docker compose.
```
Comment (judell) [inline docs/cloud/integrations/apachesuperset.md:37]:
```suggestion
[Apache Superset](https://superset.apache.org/docs/installation/) is a free to use open-source application that runs in a container, or on the desktop. Here we run it locally using docker compose.
```

File: docs/cloud/integrations/apachesuperset.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+Alternatively, you can connect to the database through the `SQLALCHEMY URI` option and use the connection String.
+
+<div style={{"marginTop":"1em", "marginBottom":"1em", "width":"90%"}}>
+<img src="/images/docs/cloud/apache-superset-uri-connection.png" />
+</div>
+
+Now add a `Dataset` which is a representation of a table you want exposed for querying. To do that click on `+Dataset` from the Datasets tab and update the database, schema and table details. Here we add a dataset using the `azure_compute_disk` table to list the Azure compute disks.
+
+<div style={{"marginTop":"1em", "marginBottom":"1em", "width":"50%"}}>
```
Comment (judell) [inline docs/cloud/integrations/apachesuperset.md:53]: crop out the whitespace in the image?

File: docs/cloud/integrations/apachesuperset.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+To create the table, select the `azure_compute_disk` dataset, set the `QUERY MODE` to `RAW RECORDS`, update the columns to be displayed and click `CREATE CHART`. Superset will then preview the data.
+
+<div style={{"marginTop":"1em", "marginBottom":"1em", "width":"90%"}}>
+<img src="/images/docs/cloud/apache-superset-disk-preview.png" />
+</div>
+
+## Create charts with custom queries.
+
+You can augment your SQL queries through Supersets `SQL Lab` under the SQL tab and visualize it with a suite of charts. To try that, click `SQL Lab` and paste this SQL that returns the number of compute disks region wise.
```
Comment (judell) [inline docs/cloud/integrations/apachesuperset.md:65]:
```suggestion
You can augment your SQL queries using Superset's `SQL Lab` (under the SQL tab) and visualize them with a suite of charts. To try that, click `SQL Lab` and paste this SQL that returns the number of compute disks per region.
```

File: docs/cloud/integrations/apachesuperset.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  region;
+  ```
+
+Once the result is returned, click `CREATE CHART` and select `PIE CHART`. Now Update the region and count columns into the dimensions field, set the metric field to aggregate `MAX` count and click `UPDATE CHART`. Custom queries can also be saved as datasets.
+
+<div style={{"marginTop":"1em", "marginBottom":"1em", "width":"90%"}}>
+<img src="/images/docs/cloud/apache-superset-custom-query.png" />
+</div>
+
+## Create dashboard to analyze resources
```
Comment (judell) [inline docs/cloud/integrations/apachesuperset.md:83]:
```suggestion
## Create a dashboard to analyze resources
```

File: docs/cloud/integrations/apachesuperset.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+Charts that display data are the building blocks of Apache Superset dashboards. We'll focus here on a creating a dashboard that monitors the Azure VMs and disks. To being, click `+Dashboard` from the Dashboards tab, and click `+Create a New Chart`. Then Create datasets and charts with `azure_compute_virtual_machine_metric_cpu_utilization_hourly` and `azure_compute_disk_metric_read_ops_daily` tables. Finally paste these queries to create dataset charts for VMs with disks not encrypted by customer managed key and list of unattached disks.
```
Comment (judell) [inline docs/cloud/integrations/apachesuperset.md:85]:
```suggestion
Charts that display data are the building blocks of Apache Superset dashboards. We'll focus here on a creating a dashboard that monitors the Azure VMs and disks. To begin, click `+Dashboard` from the Dashboards tab, and click `+Create a New Chart`. Then create datasets and charts with `azure_compute_virtual_machine_metric_cpu_utilization_hourly` and `azure_compute_disk_metric_read_ops_daily` tables. Finally paste these queries to create dataset charts for VMs with disks not encrypted by customer-managed key, and for unattached disks.
```

File: docs/cloud/integrations/apachesuperset.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  time_created,
+  encryption_type,
+  network_access_policy
+from
+  azure_compute_disk
+where
+  disk_state = 'Unattached';
+  ```
+
+Superset provides a variety of chart types. Here we use `Bar Chart` for azure_compute_virtual_machine_metric_cpu_utilization_hourly, `Funnel Chart` for azure_compute_disk_metric_read_ops_daily, `Table` for list of unattached disks and `Gauge Chart` for VMs with disks not encrypted by customer managed key. The dashboard can also be saved as an image by simply clicking on `Download as image` from the dashboard menu.
```
Comment (judell) [inline docs/cloud/integrations/apachesuperset.md:111]:
```suggestion
Superset provides a variety of chart types. Here we use `Bar Chart` for `azure_compute_virtual_machine_metric_cpu_utilization_hourly`, `Funnel Chart` for `azure_compute_disk_metric_read_ops_daily`, `Table` for list of unattached disks and `Gauge Chart` for VMs with disks not encrypted by customer-managed key. The dashboard can also be saved as an image by clicking on `Download as image` from the dashboard menu.
```

Comment (judell) [reply docs/cloud/integrations/apachesuperset.md:111]: Nice mix of charts!

File: docs/cloud/integrations/apachesuperset.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+<div style={{"marginTop":"1em", "marginBottom":"1em", "width":"90%"}}>
+<img src="/images/docs/cloud/apache-superset-dashboard-preview.png" />
+</div>
+
+## Summary
+
+With Apache Superset and Steampipe Cloud you can:
+
+- Create interactive dashboards driven by data from the tables and queries in your Steampipe Cloud workspace
```
Comment (judell) [inline docs/cloud/integrations/apachesuperset.md:121]:
```suggestion

- View tables in your Steampipe Cloud workspace

- Write custom queries for those tables

- Create interactive dashboards driven by your tables and queries
```

**Review by judell — CHANGES_REQUESTED** [review]

File: docs/cloud/integrations/apachesuperset.md (hunk 1 above)
Comment (RahulSrivastav14) [reply docs/cloud/integrations/apachesuperset.md:37]: Updated the suggested text

File: docs/cloud/integrations/apachesuperset.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+[Apache Superset](https://superset.apache.org/docs/installation/) is a free to use open-source application that runs in a container, or on the desktop. Here we run it locally through docker compose.
+
+With Apache Superset up and running, open a browser to `http://localhost:8080/`. Click on `Database Connections` under settings on the top right corner, then click `+Database` and select `PostgreSQL`. Enter the connection details, toggle the SSL mode to `Require` and click connect.
+
+<div style={{"marginTop":"1em", "marginBottom":"1em", "width":"90%"}}>
+<img src="/images/docs/cloud/apache-superset-connection-success.png" />
+</div>
+
+Alternatively, you can connect to the database through the `SQLALCHEMY URI` option and use the connection String.
```
[the remark this answers is not in the document]
Comment (RahulSrivastav14) [reply docs/cloud/integrations/apachesuperset.md:45]: Rectified the text

File: docs/cloud/integrations/apachesuperset.md (hunk 3 above)
Comment (RahulSrivastav14) [reply docs/cloud/integrations/apachesuperset.md:65]: updated suggested text

File: docs/cloud/integrations/apachesuperset.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+select
+  region,
+  count(name)
+from
+  azure_compute_disk
+group by
+  region;
+  ```
+
+Once the result is returned, click `CREATE CHART` and select `PIE CHART`. Now Update the region and count columns into the dimensions field, set the metric field to aggregate `MAX` count and click `UPDATE CHART`. Custom queries can also be saved as datasets.
```
[the remark this answers is not in the document]
Comment (RahulSrivastav14) [reply docs/cloud/integrations/apachesuperset.md:77]: rectified

File: docs/cloud/integrations/apachesuperset.md (hunk 4 above)
Comment (RahulSrivastav14) [reply docs/cloud/integrations/apachesuperset.md:83]: Rectified

File: docs/cloud/integrations/apachesuperset.md (hunk 6 above)
Comment (RahulSrivastav14) [reply docs/cloud/integrations/apachesuperset.md:111]: Thank you. Wanted to focus on just one service but multiple features

File: docs/cloud/integrations/apachesuperset.md (hunk 7 above)
Comment (RahulSrivastav14) [reply docs/cloud/integrations/apachesuperset.md:121]: Updated suggested text

**Review by judell — APPROVED** [review]
LGTM

**Pull request merged**